modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

```diff
@@ -0,0 +1,260 @@
+
+----
+import * from jsonschema!example::schema::Person
+----
+
+== Use Your Types in a DataWeave Script
+
+Having the import directive from the above example in the script header loads all the existing types in the JSON schema. In `import * from jsonschema!example::schema::Person`, the only existing type is the `Root` type, specified at the root which describes an Object that can have three properties (`firstName`, `lastName`, and `age`). It is equivalent to declaring the following type in your DataWeave script:
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+type Root = { firstName: String, lastName: String, age?: Number }
+----
+
+Notice that the `age` field is the only one that is optional, as indicated by the `?`.
+
+You can use the type to know if a value follows the structure defined by the JSON schema.
+The following example outputs the value `true` because the object contains the required fields, `firstName` and `lastName`:
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ lastName: "Doe"
+} is Root
+----
+
+.Output:
+----
+"true"
+----
+
+The following example outputs the value `false` since the object does not contain the required fields, `firstName` and `lastName`:
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ age: 20
+} is Root
+----
+
+.Output:
+----
+"false"
+----
+
+== Use Definitions Inside Schemas
+
+Definitions inside JSON schemas are also loaded when the module is imported:
+
+.JSON Schema (`example/schema/Account.json`):
+[source,JSON]
+----
+{
+ "$schema": "http://json-schema.org/draft-04/schema#",
+ "definitions": {
+   "address": {
+     "type": "object",
+     "properties": {
+       "street_address": {
+         "type": "string"
+       },
+       "city": {
+         "type": "string"
+       },
+       "state": {
+         "type": "string"
+       }
+     },
+     "required": [
+       "street_address",
+       "city",
+       "state"
+     ]
+   },
+   "person": {
+     "type": "object",
+     "properties": {
+       "firstName": {
+         "type": "string",
+         "description": "The person's first name."
+       },
+       "lastName": {
+         "type": "string",
+         "description": "The person's last name."
+       },
+       "age": {
+         "description": "Age in years which must be equal to or greater than zero.",
+         "type": "integer",
+         "minimum": 0
+       }
+     },
+     "required": [
+       "firstName",
+       "lastName"
+     ]
+   }
+ },
+ "type": "object",
+ "properties": {
+   "person": {
+     "$ref": "#/definitions/person"
+   },
+   "address": {
+     "$ref": "#/definitions/address"
+   }
+ }
+}
+----
+
+You can import the types from that schema with the following directive, assuming the JSON schema is `example/schema/Account.json`:
```

Comment:
The "assuming..." bit is a little awkward.
Maybe...
For example, to import the types ...

Or

If the schema (file?) is xxx, use the following directive...

Or maybe you don't need that bit starting with ", assuming" because it's already in the header?

Reply:
Reword the sentence by deleting the _assuming_ part.